opencti-platform/opencti-front/src/private/components/analysis/Reports.tsx

```diff
@@ -0,0 +1,321 @@
+import React, { FunctionComponent, SyntheticEvent, useState } from 'react';
+import * as R from 'ramda';
+import { Record } from 'mdi-material-ui';
+import { QueryRenderer } from '../../../relay/environment';
+import { convertFilters } from '../../../utils/ListParameters';
+import ListLines from '../../../components/list_lines/ListLines';
+import ReportsLines, { reportsLinesQuery } from './reports/ReportsLines';
+import ReportCreation from './reports/ReportCreation';
+import ToolBar from '../data/ToolBar';
+import Security from '../../../utils/Security';
+import { KNOWLEDGE_KNUPDATE } from '../../../utils/hooks/useGranted';
+import { UserContext } from '../../../utils/hooks/useAuth';
+import useLocalStorage from '../../../utils/hooks/useLocalStorage';
+import { ReportsLinesPaginationQuery$data } from './reports/__generated__/ReportsLinesPaginationQuery.graphql';
+import { Filters } from '../../../components/list_lines';
+import { ModuleHelper } from '../../../utils/platformModulesHelper';
+import { ReportLine_node$data } from './reports/__generated__/ReportLine_node.graphql';
+
+const LOCAL_STORAGE_KEY = 'view-reports';
+
+interface ReportsProps {
+  objectId: string,
+  authorId: string,
+  onChangeOpenExports: () => void,
+  match: { params: { reportType: string } },
+}
+
+const Reports: FunctionComponent<ReportsProps> = ({
+  objectId,
+  authorId,
+  onChangeOpenExports,
+  match,
+}) => {
+  const [viewStorage, setViewStorage, storageHelpers] = useLocalStorage(LOCAL_STORAGE_KEY, {
+    numberOfElements: { number: 0, symbol: '' },
+    filters: {} as Filters,
+    searchTerm: '',
+    sortBy: 'published',
+    orderAsc: false,
+    openExports: false,
+    count: 25,
+    view: 'lines',
+  });
+
+  const {
+    numberOfElements,
+    filters,
+    searchTerm,
+    sortBy,
+    orderAsc,
+    openExports,
+    view,
+  } = viewStorage;
+
+  const [selectedElements, setSelectedElements] = useState<Record<string, ReportLine_node$data>>({});
+  const [deSelectedElements, setDeSelectedElements] = useState<Record<string, ReportLine_node$data>>({});
+  const [selectAll, setSelectAll] = useState(false);
+  const [NotEqLocalFilterMode] = useState('and');
+  const [EqLocalFilterMode] = useState('or');
+
+  const { reportType } = match.params;
```

Comment:
Use a hook

Reply:
to be done when doing the issue on AND/Or operators switching

Reply:
Why is that ? Why can't you do it in this task ?